magma/tuple.py

```diff
@@ -131,6 +131,10 @@ def iswhole(self, ts):
             if ts[i].name.index != self.Ks[i]:
                 return False
 
+        if len(ts) != len(ts[0].name.tuple):
```

Comment:
It seems like this "exclusion" method is not robust (e.g. we found another case that indicated non-whole-ness)...probably would be better to write down the exact condition we want and check for it.

But if this fixes the immediate issue, we can fix ^ separately.

Reply:
Agreed